usaspending_api/agency/v2/views/award_count.py

```diff
@@ -0,0 +1,120 @@
+    """
+    Obtain the count of awards for a specific agency in a single fiscal year.
+    """
+
+    endpoint_doc = "usaspending_api/api_contracts/contracts/v2/agency/awards/count.md"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.params_to_validate = ["fiscal_year"]
+        self._toptier_awarding_agencies_to_include = [
+            "Agency for International Development",
+            "Department of Agriculture",
+            "Department of Commerce",
+            "Department of Defense",
+            "Department of Education",
+            "Department of Energy",
+            "Department of Health and Human Services",
+            "Department of Homeland Security",
+            "Department of Housing and Urban Development",
+            "Department of the Interior",
+            "Department of Justice",
+            "Department of Labor",
+            "Department of State",
+            "Department of Transportation",
+            "Department of the Treasury",
+            "Department of Veterans Affairs",
+            "Environmental Protection Agency",
+            "General Services Administration",
+            "National Aeronautics and Space Administration",
+            "National Science Foundation",
+            "Nuclear Regulatory Commission",
+            "Office of Personnel Management",
+            "Small Business Administration",
+            "Social Security Administration",
+        ]
+
+    @cache_response()
+    def get(self, request: Request, *args: Any, **kwargs: Any) -> Response:
+        self._fy_end = get_fiscal_year_end_datetime(self.fiscal_year).date()
+        self._fy_start = get_fiscal_year_start_datetime(self.fiscal_year).date()
+
+        results = self.query_elasticsearch_for_prime_awards({})
+
+        raw_response = {"results": results, "messages": []}
+
+        return Response(raw_response)
+
+    def _generate_filters(self, awarding_toptier_agency_name):
+        return {
+            "agencies": [{"type": "awarding", "tier": "toptier", "name": awarding_toptier_agency_name}],
+            "time_period": [{"start_date": self._fy_start, "end_date": self._fy_end}],
+        }
+
+    def query_elasticsearch_for_prime_awards(self, filters) -> list:
+        toptier_awarding_agencies_to_include_results = []
+        filter_options = {}
+        time_period_obj = AwardSearchTimePeriod(
+            default_end_date=settings.API_MAX_DATE, default_start_date=settings.API_SEARCH_MIN_DATE
+        )
+        new_awards_only_decorator = NewAwardsOnlyTimePeriod(
+            time_period_obj=time_period_obj, query_type=_QueryType.AWARDS
+        )
+        filter_options["time_period_obj"] = new_awards_only_decorator
+        for awarding_toptier_agency_name in self._toptier_awarding_agencies_to_include:
+            awarding_toptier_agency_code = (
+                ToptierAgency.objects.filter(name=awarding_toptier_agency_name).first().toptier_code
```

Comment:
Instead of querying toptier agency for each record in the `toptier_awarding_agencies_to_include_results` list, could this be refactored to use a single query? 

Alternatively, why not update the hardcoded list to include codes instead of names to avoid the query altogether?

Reply:
Never mind about the second suggestion. I see that the code is only used for the final response, not for the ES query.